src/Mechanisms/HandleComponents/HandleComponents.php

```diff
@@ -412,7 +416,8 @@ protected function recursivelySetValue($baseProperty, $target, $leafValue, $segm
     protected function setComponentPropertyAwareOfTypes($component, $property, $value)
     {
         try {
-           $component->$property = $value;
+            // If the property is getting removed, unset it (making it null)
+            $component->$property = $value;
```

Comment:
Sorry, is the comment applicable here? Seems like we're "setting" the property on line 420, but the comment says "unset it"

Maybe you're just saying that an __rm__'d value is being unset by it being set as null earlier in the codepath?

maybe a more descriptive comment would be helpful?

Reply:
Oh, that's a remnant comment from an earlier revision. I'll clean it up.